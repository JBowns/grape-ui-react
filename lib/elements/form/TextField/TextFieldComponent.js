'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TextFieldComponent = undefined;

var _templateObject = _taggedTemplateLiteral(['\n    ', '\n    ', '\n  '], ['\n    ', '\n    ', '\n  ']);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _styledComponents = require('styled-components');

var _styledComponents2 = _interopRequireDefault(_styledComponents);

var _textStyles = require('../../../elements/typography/textStyles');

var _baseControlStyle = require('../ControlGroup/baseControlStyle');

var _componentHelpers = require('../../../utils/componentHelpers');

var _styledSystem = require('styled-system');

var _decorators = require('../../../decorators');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _taggedTemplateLiteral(strings, raw) { return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var TextFieldComponent = exports.TextFieldComponent = function TextFieldComponent(props) {
  var textBase = (0, _textStyles.textStylesBase)(props);
  var controlBase = (0, _baseControlStyle.controlStylesBase)(props);
  var ProtoBase = _styledComponents2.default.input.attrs({
    type: 'text'
  })(_templateObject, textBase, controlBase);
  var ProtoStyledSystem = (0, _decorators.withStyledSystem)(ProtoBase, props);
  return _react2.default.createElement(ProtoStyledSystem, (0, _componentHelpers.passThrough)(TextFieldComponent, props));
};

TextFieldComponent.propTypes = Object.assign({
  allowStyledWidth: _propTypes2.default.bool,
  borderColor: _propTypes2.default.string,
  borderColorActive: _propTypes2.default.string,
  borderRadius: _propTypes2.default.string,
  color: _propTypes2.default.string,
  disallowAllStyled: _propTypes2.default.bool,
  fontFamily: _propTypes2.default.string,
  fontWeight: _propTypes2.default.string,
  kerning: _propTypes2.default.string,
  lg: _propTypes2.default.bool,
  padding: _propTypes2.default.string,
  sm: _propTypes2.default.bool,
  textAlign: _propTypes2.default.string,
  textDecoration: _propTypes2.default.string
}, _styledSystem.width.propTypes);

TextFieldComponent.defaultProps = {
  color: _textStyles.defaultTextStylesBase.color,
  fontFamily: _textStyles.defaultTextStylesBase.fontFamily,
  fontWeight: _textStyles.defaultTextStylesBase.fontWeight,
  kerning: _textStyles.defaultTextStylesBase.kerning,
  lg: _textStyles.defaultTextStylesBase.lg,
  sm: _textStyles.defaultTextStylesBase.sm,
  textAlign: _textStyles.defaultTextStylesBase.textAlign,
  textDecoration: _textStyles.defaultTextStylesBase.textDecoration,
  borderColor: _baseControlStyle.defaultControlStylesBase.borderColor,
  borderColorActive: _baseControlStyle.defaultControlStylesBase.borderColorActive,
  borderRadius: _baseControlStyle.defaultControlStylesBase.borderRadius,
  padding: _baseControlStyle.defaultControlStylesBase.padding,
  disallowAllStyled: true,
  allowStyledWidth: true,
  width: 1
};