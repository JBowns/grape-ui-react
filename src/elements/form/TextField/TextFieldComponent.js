import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { defaultTextStylesBase, textStylesBase } from 'src/elements/typography/textStyles';
import { defaultControlStylesBase, controlStylesBase } from '../ControlGroup/baseControlStyle';
import { passThrough } from 'src/utils/componentHelpers';
import { width } from 'styled-system';
import { withStyledSystem } from 'src/decorators';

export const TextFieldComponent = props => {
  const textBase = textStylesBase(props);
  const controlBase = controlStylesBase(props);
  const ProtoBase = styled.input.attrs({
    type: 'text',
  })`
    ${textBase}
    ${controlBase}
  `;
  const ProtoStyledSystem = withStyledSystem(ProtoBase, props);
  return <ProtoStyledSystem {...passThrough(TextFieldComponent, props)} />;
};

TextFieldComponent.propTypes = {
  allowStyledWidth: PropTypes.bool,
  borderColor: PropTypes.string,
  borderColorActive: PropTypes.string,
  borderRadius: PropTypes.string,
  color: PropTypes.string,
  disallowAllStyled: PropTypes.bool,
  fontFamily: PropTypes.string,
  fontWeight: PropTypes.string,
  kerning: PropTypes.string,
  lg: PropTypes.bool,
  padding: PropTypes.string,
  sm: PropTypes.bool,
  textAlign: PropTypes.string,
  textDecoration: PropTypes.string,
  ...width.propTypes,
};

TextFieldComponent.defaultProps = {
  color: defaultTextStylesBase.color,
  fontFamily: defaultTextStylesBase.fontFamily,
  fontWeight: defaultTextStylesBase.fontWeight,
  kerning: defaultTextStylesBase.kerning,
  lg: defaultTextStylesBase.lg,
  sm: defaultTextStylesBase.sm,
  textAlign: defaultTextStylesBase.textAlign,
  textDecoration: defaultTextStylesBase.textDecoration,
  borderColor: defaultControlStylesBase.borderColor,
  borderColorActive: defaultControlStylesBase.borderColorActive,
  borderRadius: defaultControlStylesBase.borderRadius,
  padding: defaultControlStylesBase.padding,
  disallowAllStyled: true,
  allowStyledWidth: true,
  width: 1,
};
