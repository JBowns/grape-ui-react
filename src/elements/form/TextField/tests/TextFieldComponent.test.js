import React from 'react';
import { TextFieldComponent } from '../TextFieldComponent';
import 'jest-styled-components';
// import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('TextFieldComponent base', () => {
  it('should return a TextFieldComponent object', () => {
    const component = mount(<TextFieldComponent />);
    expect(toJson(component.find('TextFieldComponent'))).toMatchSnapshot();
  });
  it('should return a small TextFieldComponent object', () => {
    const component = mount(<TextFieldComponent sm />);
    expect(toJson(component.find('TextFieldComponent'))).toMatchSnapshot();
  });
  it('should return a large TextFieldComponent object', () => {
    const component = mount(<TextFieldComponent lg />);
    expect(toJson(component.find('TextFieldComponent'))).toMatchSnapshot();
  });
});

describe('TextField with a responsive width', () => {
  it('should return a TextFieldComponent object with a responsive width', () => {
    const component = mount(<TextFieldComponent controlId="exampleFullName" controlLabel="Full name" width={[1, 1 / 2, 1 / 3, 1 / 4]} />);
    expect(toJson(component.find('TextFieldComponent'))).toMatchSnapshot();
  });
});
