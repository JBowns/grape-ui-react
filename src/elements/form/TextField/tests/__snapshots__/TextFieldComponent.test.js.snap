// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField with a responsive width should return a TextFieldComponent object with a responsive width 1`] = `
.c0 {
  font-family: inherit;
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  border: 1px solid hsl(0,0%,25%);
  border-radius: 4px;
  padding: 1rem;
  outline: 0;
  width: 100%;
}

.c0:focus {
  border-color: hsl(323.3,84.6%,28%);
  box-shadow: 0 0 0 1px hsl(323.3,84.6%,28%);
}

@media screen and (min-width:40em) {
  .c0 {
    width: 50%;
  }
}

@media screen and (min-width:52em) {
  .c0 {
    width: 33.33333333333333%;
  }
}

@media screen and (min-width:64em) {
  .c0 {
    width: 25%;
  }
}

<TextFieldComponent
  allowStyledWidth={true}
  borderColor="hsl(0, 0%, 25%)"
  borderColorActive="hsl(323.3, 84.6%, 28%)"
  borderRadius="4px"
  color="inherit"
  controlId="exampleFullName"
  controlLabel="Full name"
  disallowAllStyled={true}
  fontFamily="inherit"
  fontWeight="inherit"
  kerning="inherit"
  lg={false}
  padding="1rem"
  sm={false}
  textAlign="inherit"
  textDecoration="inherit"
  width={
    Array [
      1,
      0.5,
      0.3333333333333333,
      0.25,
    ]
  }
>
  <styled.input
    controlId="exampleFullName"
    controlLabel="Full name"
  >
    <input
      className="c0"
      type="text"
    />
  </styled.input>
</TextFieldComponent>
`;

exports[`TextFieldComponent base should return a TextFieldComponent object 1`] = `
.c0 {
  font-family: inherit;
  font-size: inherit;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  border: 1px solid hsl(0,0%,25%);
  border-radius: 4px;
  padding: 1rem;
  outline: 0;
  width: 100%;
}

.c0:focus {
  border-color: hsl(323.3,84.6%,28%);
  box-shadow: 0 0 0 1px hsl(323.3,84.6%,28%);
}

<TextFieldComponent
  allowStyledWidth={true}
  borderColor="hsl(0, 0%, 25%)"
  borderColorActive="hsl(323.3, 84.6%, 28%)"
  borderRadius="4px"
  color="inherit"
  disallowAllStyled={true}
  fontFamily="inherit"
  fontWeight="inherit"
  kerning="inherit"
  lg={false}
  padding="1rem"
  sm={false}
  textAlign="inherit"
  textDecoration="inherit"
  width={1}
>
  <styled.input>
    <input
      className="c0"
      type="text"
    />
  </styled.input>
</TextFieldComponent>
`;

exports[`TextFieldComponent base should return a large TextFieldComponent object 1`] = `
.c0 {
  font-family: inherit;
  font-size: 1.2rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  border: 1px solid hsl(0,0%,25%);
  border-radius: 8px;
  padding: 1rem;
  outline: 0;
  width: 100%;
}

.c0:focus {
  border-color: hsl(323.3,84.6%,28%);
  box-shadow: 0 0 0 1px hsl(323.3,84.6%,28%);
}

<TextFieldComponent
  allowStyledWidth={true}
  borderColor="hsl(0, 0%, 25%)"
  borderColorActive="hsl(323.3, 84.6%, 28%)"
  borderRadius="4px"
  color="inherit"
  disallowAllStyled={true}
  fontFamily="inherit"
  fontWeight="inherit"
  kerning="inherit"
  lg={true}
  padding="1rem"
  sm={false}
  textAlign="inherit"
  textDecoration="inherit"
  width={1}
>
  <styled.input>
    <input
      className="c0"
      type="text"
    />
  </styled.input>
</TextFieldComponent>
`;

exports[`TextFieldComponent base should return a small TextFieldComponent object 1`] = `
.c0 {
  font-family: inherit;
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  color: inherit;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  border: 1px solid hsl(0,0%,25%);
  border-radius: 2px;
  padding: 1rem;
  outline: 0;
  width: 100%;
}

.c0:focus {
  border-color: hsl(323.3,84.6%,28%);
  box-shadow: 0 0 0 1px hsl(323.3,84.6%,28%);
}

<TextFieldComponent
  allowStyledWidth={true}
  borderColor="hsl(0, 0%, 25%)"
  borderColorActive="hsl(323.3, 84.6%, 28%)"
  borderRadius="4px"
  color="inherit"
  disallowAllStyled={true}
  fontFamily="inherit"
  fontWeight="inherit"
  kerning="inherit"
  lg={false}
  padding="1rem"
  sm={true}
  textAlign="inherit"
  textDecoration="inherit"
  width={1}
>
  <styled.input>
    <input
      className="c0"
      type="text"
    />
  </styled.input>
</TextFieldComponent>
`;
